frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx

```diff
@@ -0,0 +1,255 @@
+          kind: VolumeSnapshotModel.kind,
+          apiGroup: VolumeSnapshotModel.apiGroup,
+        },
+        accessModes,
+        resources: {
+          requests: {
+            storage: `${requestedSize}${requestedUnit}`,
+          },
+        },
+      },
+    };
+
+    // eslint-disable-next-line promise/catch-or-return
+    handlePromise(k8sCreate(PersistentVolumeClaimModel, restorePVCTemplate, namespace)).then(
+      (newPVC) => {
+        close();
+        history.push(resourceObjPath(newPVC, referenceFor(newPVC)));
+      },
+    );
+  };
+
+  return (
+    <div className="modal-content modal-content--no-inner-scroll">
+      <ModalTitle>Restore as new PVC</ModalTitle>
+      <Form onSubmit={submit} name="form">
+        <ModalBody>
+          <p>When restore action is finished a new crash-consistent PVC copy will be created.</p>
+          <FormGroup
+            label="Name"
+            isRequired
+            fieldId="pvc-name"
+            className="co-restore-pvc-modal__input"
+          >
+            <TextInput
+              isRequired
+              type="text"
+              id="pvc-name"
+              name="restore-pvc-modal__name"
+              value={restorePVCName}
+              onChange={setPVCName}
+            />
+          </FormGroup>
+          <FormGroup
+            fieldId="restore-storage-class"
+            className="co-restore-pvc-modal__input"
+            isRequired
+          >
+            {progress ? (
+              <LoadingInline />
+            ) : (
+              <StorageClassDropdown
+                onChange={handleStorageClass}
+                filter={onlyPvcSCs}
+                id="restore-storage-class"
+                required
+              />
+            )}
+          </FormGroup>
+          <FormGroup
+            label="Size"
+            isRequired
+            fieldId="pvc-size"
+            className="co-restore-pvc-modal__input"
+          >
+            <RequestSizeInput
+              name="requestSize"
+              onChange={requestedSizeInputChange}
+              defaultRequestSizeUnit={requestedUnit}
+              defaultRequestSizeValue={requestedSize}
+              dropdownUnits={dropdownUnits}
+              required
+            />
+          </FormGroup>
+          <div className="co-restore-pvc-modal__details-section">
+            <p className="text-muted">{VolumeSnapshotModel.label} details</p>
+            <Grid hasGutter>
+              <GridItem span={6}>
+                <div className="co-restore-pvc-modal__pvc-details">
+                  <strong>Created At</strong>
+                  <span>
+                    <Timestamp timestamp={resource?.metadata?.creationTimestamp} />
+                  </span>
+                </div>
+                <div className="co-restore-pvc-modal__pvc-details">
+                  <strong>Status</strong>
+                  <Status status={resource?.status?.readyToUse ? 'Ready' : 'Not Ready'} />
```

Comment:
Is there a stage where readyToUse is not populated?

Reply:
Mostly the status is populated right after the creation, but its good to check so that it doesn't break the view.